tools/megengine

```diff
@@ -17,6 +17,7 @@ sync() {
     curl --silent --location --output "./third_party/source/megengine/src/serialization/fbs/opr_param_defs.fbs" "https://github.com/MegEngine/MegEngine/blob/master/ci/compatibility/fbs/V2-backup/opr_param_defs.fbs?raw=true"
     curl --silent --location --output "./third_party/source/megengine/src/serialization/fbs/schema_v2.fbs" "https://github.com/MegEngine/MegEngine/blob/master/ci/compatibility/fbs/V2-backup/schema_v2.fbs?raw=true"
     sed -i 's/struct/table/g; s/\<DTypeEnum\>/fbs.DTypeEnum/g; s/\<DType\>/fbs.DType/g; s/\[ubyte:4\]/ubyte/g; s/(bit_flags)//g; s/table fbs.DType/table DType/g; s/table fbs.DTypeEnum/table DTypeEnum/g' ./third_party/source/megengine/src/serialization/fbs/*[^dtype].fbs
+    sed -i 's/items\:\[IndexDescMaskItem\]/items\:IndexDescMaskItem/g; s/desc\:\[AxisDesc\]/desc\:AxisDesc/g' ./third_party/source/megengine/src/serialization/fbs/mgb_cpp_opr.fbs
```

Comment:
Can you provide some context why these changes are needed? Should this get fixed in the `.fbs` file?

Reply:
In the process of user visualization, we frequently find that there is a boundary crossing problem(i.e. `Offset is outside the bounds of the DataView`). After debugging, we find that the problem lies in these two params and is related to our dump process. Only the number of params in the dump, but not the actual content.
Since the user is using it and the information of these two params(i.e. IndexDescMaskDump and AxisAddRemove) is irrelevant, we have temporarily adjusted it so that the user no longer feels the problem. Due to the arrangement of work, we will conduct further investigation later.
Here is an example of minimal replication.

Reply:
Can you please investigate. Are the files corrupted or does the issue reproduce during runtime execution? Rewriting file schemas is problematic as these issues can't be addressed later without breaking compatibility.

Reply:
ok

Reply:
The problem is `sed -i 's/struct/table/g' ./third_party/source/megengine/src/serialization/fbs/*[^dtype].fbs` in `tools/megengine sync`. When using `tools/flatc.js` to generate `schema.js`, the deserialization method of `struct` should be generated instead of `table`.

When there is `struct`, I find that there is a problem when using `tools/flatc.js` to generate `schema.js`, as follows:
```javascript
// test.fbs
namespace mgb.serialization.fbs;

enum AxisDescMethod : byte {
    ADD_1,
    REMOVE,
}

struct AxisDesc {
    method:AxisDescMethod;  // change to method:byte;
    axis:int;
}

table AxisAddRemove {
    desc:[AxisDesc];
}

root_type AxisAddRemove;
```
For the `test.fbs` file above, `tools/flatc.js` can't work normally, However, if the `method:AxisDescMethod` field in `struct AxisDesc` is changed to `method:byte` (as shown in the following code block), `tools/flatc.js` can work normally. Is it because the generation cannot be nested? Can you help me to look at this problem?

```javascript
// test.fbs
namespace mgb.serialization.fbs;

enum AxisDescMethod : byte {
    ADD_1,
    REMOVE,
}

struct AxisDesc {
    method:byte;  // change from method:AxisDescMethod;
    axis:int;
}

table AxisAddRemove {
    desc:[AxisDesc];
}

root_type AxisAddRemove;
```

Reply:
When there is `struct`, I find that there is a problem when using `tools/flatc.js` to generate `schema.js`, as follows:
```javascript
// test.fbs
namespace mgb.serialization.fbs;

enum AxisDescMethod : byte {
    ADD_1,
    REMOVE,
}

struct AxisDesc {
    method:AxisDescMethod;  // change to method:byte;
    axis:int;
}

table AxisAddRemove {
    desc:[AxisDesc];
}

root_type AxisAddRemove;
```
For the `test.fbs` file above, `tools/flatc.js` can't work normally, However, if the `method:AxisDescMethod` field in `struct AxisDesc` is changed to `method:byte` (as shown in the following code block), `tools/flatc.js` can work normally. Is it because the generation cannot be nested? Can you please help me to look at this problem?

```javascript
// test.fbs
namespace mgb.serialization.fbs;

enum AxisDescMethod : byte {
    ADD_1,
    REMOVE,
}

struct AxisDesc {
    method:byte;  // change from method:AxisDescMethod;
    axis:int;
}

table AxisAddRemove {
    desc:[AxisDesc];
}

root_type AxisAddRemove;
```